File: EnterLeaveRoom/EnterLeaveRoom.js
const calculateEnterLeaveRoom = (reqString) => {
    let maxEnterLeave = reqString.length
    let minEnterLeave = 0
    let ans = 0

    for (let i=0; i<reqString.length; i++) {
        if (reqString[i] === '1')
            minEnterLeave ++;
        else if (reqString[i] === '0')
            minEnterLeave --;

        if (minEnterLeave < 0) {
            ans ++
            minEnterLeave = 0
        }
        minEnterLeave = Math.max(minEnterLeave, ans)
    }
    console.log("Max: ", maxEnterLeave)
    console.log("Min: ", minEnterLeave)
    
}

calculateEnterLeaveRoom("010")

File: EnterLeaveRoom/EnterLeaveRoom.java
public class EnterLeaveRoom {

    public void calculateEnterLeaveRoom(String reqString) {
        int i = 0;
        
        int maxEnterLeave = reqString.length();
        int minEnterLeave = 0;
        int ans = 0;
        for (i=0; i<reqString.length(); i++) {
            if(reqString.charAt(i) == '1') {
                minEnterLeave++;
            }
            else if(reqString.charAt(i) == '0') {
                minEnterLeave--;
            }
           
            if(minEnterLeave < 0) {
                minEnterLeave = 0;
                ans += 1;
            }
            ans = Math.max(minEnterLeave, ans);

        }

        System.out.println("Max: " + maxEnterLeave);
        System.out.println("Min: " + ans);

    }

    public static void main(String[] args) {

        EnterLeaveRoom e = new EnterLeaveRoom();

        e.calculateEnterLeaveRoom("010");
    }
}
